pkg/storage/postgresql/findings.go

```diff
@@ -249,6 +264,8 @@ func (db DB) ListFindings(filter storage.Filter) ([]model.FindingExpanded, stora
 					{{ end }}
 				{{ end }}
 				{{ if and (eq .Status "OPEN") (or .MinDate .MaxDate) }}
+					/* Ignore findings marked as false positive */
+					AND f.status != 'FALSE_POSITIVE'
 					{{ if .MinDate }}
 					AND fexp.found_at >= :minDate
 					{{ end }}
```

Comment:
I think there is a tricky case here for the `AtDate` "branch".
For the case of using `minDate` or `maxDate` filters, then we ignore findings that have been marked as `FALSE POSITIVE`, ok, I think that is correct, so we don't count them as if they have ever been "opened" nor "fixed".
For the case of using `atDate` filter with a `status` filter, we also ignore FALSE POSITIVE findings as if they were never "opened" nor "fixed".
But, for the case of using `atDate` filter without status filter, I think we are not ignoring these findings, so they will be taken into consideration and their status will be calculated based on their `finding_exposures table` data.

So I think we should ignore `FALSE POSITIVE` findings also in the `{{ else }}` from the `{{if .Status }}` inside the `{{ if .AtDate }}`  "branch", shouldn't we? Did I misunderstood something?

Reply:
> For the case of using `atDate` filter with a `status` filter, we also ignore FALSE POSITIVE findings as if they were never "opened" nor "fixed".

We ignore FALSE POSITIVE if the status passed by the user is OPEN or FIXED. If the user sets `atDate` with `status=FALSE_POSITIVE`, then the query will return findings that were found before `:atDate` and are marked as false positive.

Keeping that in mind, I find it correct to include FALSE POSITIVE when the user performs a query with `atDate` only.

Makes sense?

Reply:
> Keeping that in mind, I find it correct to include FALSE POSITIVE when the user performs a query with atDate only.

Mmm ok, I see, that happens because we abort the recalculation of exposures and status based on them here:
https://github.com/adevinta/vulnerability-db-api/blob/a627b3559b3fdb4d1c414b0b3aec3856f2b26fd1/pkg/storage/postgresql/findings.go#L689

But this:

> If the user sets atDate with status=FALSE_POSITIVE, then the query will return findings that were found before :atDate and are marked as false positive.

Is not the case, right?
If user sets `atDate` with `status=FALSE_POSITIVE`, that status will be ignored and will return findings that comply with that atDate param for all statuses. I guess this is correct, right? For me it makes sense to not support the false positive state with the atDate param as their status and exposures will always be the same, because we are skipping the recalculation based on the queried time.

Reply:
> If user sets `atDate` with `status=FALSE_POSITIVE`, that status will be ignored and will return findings that comply with that atDate param for all statuses. I guess this is correct, right?

You are right!